### DISCUSSION TITLE
`session` in middleware returns `null` when using fetch API from server component

### DISCUSSION THREAD

**Discussion by rodzerjohn** [supabase/supabase#10456] [Questions]
Hello, thanks for newly added docs for nextjs 13 `app` directory, (https://supabase.com/docs/guides/auth/auth-helpers/nextjs-server-components) everything works fine, except i have little problem with fetch API implementation.

I'm trying to access `/api/hello` from server component using fetch API. I want that api to be protected by middleware using `supabase.auth.getSession()` but when I console.log session data it returns `null`.
In client component everything works fine, I get this problem only in server component.

Example with server component.
```
const getUser = async () => {
    const res = await fetch("http://localhost:3000/api/hello");

    if (!res?.ok) { return null; }

    return await res.json();
}

const Header = async () => {
    const user = await getUser();

    return(
        <div className="container mx-auto">
            {user}
        </div>
    )
}

export default Header
```

**Comment by hjaber** [comment]
It is difficult to debug without additional information. Could you please post your entire server component file, including your imports.  Could you also post your `api/hello` file as well as your `supabase` client file. The problem could lie in any of the previously mentioned files.

**Reply by rodzerjohn** [reply]
> It is difficult to debug without additional information. Could you please post your entire server component file, including your imports. Could you also post your `api/hello` file as well as your `supabase` client file. The problem could lie in any of the previously mentioned files.

The code I have provided is my entire server component file. Everything else i have is from here https://supabase.com/docs/guides/auth/auth-helpers/nextjs-server-components i just copied everything from docs and added little code with fetch api.
